extras/menus/advancedvsh/advanced.c

```diff
@@ -46,385 +46,416 @@ extern char* available_fonts[];
 
 int submenu_draw(void)
 {
-	u32 fc,bc;
-	const char *msg;
+	const char msg[128] = {0};
```

Comment:
Why have this static instead of dynamic ?

Reply:
Previously `*msg` was assigned to `g_messages[<option>]` and displayed as is.
That's why the satelite files had leading and trailing spaces.

I use `msg[128]` as a buffer to copy the value from `g_messages[<option>]` with the necessary padding to then display it with.
The copying and padding is done with `snprintf` to prevent any overflows.
The buffer is also long enough to accommodate any string from the satelite files, but just in case `snprintf` will stop it after 128 chars.
Also, it's just easier to have a "big" static buffer to write into than having to allocate and free every option that needs displaying based of their length and the needed padding. That just adds extra complexity and asking for memory leaks IMO.
Static array makes sense to me here.

If not then I can make an attempt at dynamically allocating the `msg` array with `vpl_alloc()` and `vpl_free()`.